python/lsst/meas/extensions/shapeHSM/_hsm_shape.py

```diff
@@ -259,20 +259,25 @@ def measure(self, record, exposure):
             # equivalent of each argument for clarity.
             # TODO: [DM-42047] Change to public API when an optimized version
             # is available.
-            galsim._galsim.EstimateShearView(
-                shape._data,  # shape data buffer (not passed in pure Python)
-                image._image,  # gal_image
-                psf._image,  # PSF_image
-                weight._image,  # weight
-                float(skyvar),  # sky_var
-                self.config.shearType.upper(),  # shear_est
-                recomputeFlux.upper(),  # recompute_flux
-                float(2.5 * psfSigma),  # guess_sig_gal
-                float(psfSigma),  # guess_sig_PSF
-                float(precision),  # precision
-                guessCentroid._p,  # guess_centroid
-                hsmparams._hsmp,  # hsmparams
-            )
+            try:
```

Comment:
You shouldn't need a `try` block within a `try` block. Just catch both types of errors in the same `except` block below. Reference for capturing two types of errors: https://docs.python.org/3/tutorial/errors.html#handling-exceptions

Reply:
Oh, you're right, that's way better!

Reply:
The point of wrapping only that one line in another try block was that the `RuntimeErrors` were guaranteed to come from GalSim C++. That's still probably going to be true for the rest of the block but not necessarily if someone adds other bits of non-GalSim code to it.

Reply:
The rest of the code in the block doesn't seem troublesome to me, but I agree with @taranu that keeping them there poses the risk of others adding non-GalSim code to it. Looking at the [original C++ code](https://github.com/lsst/meas_extensions_shapeHSM/blob/87969a01f44e02d64a0745474a40150de2db046b/src/HsmShape.cc#L191) I used for this Python implementation, it actually makes more sense to use one `try...except` block and only include GalSim's `EstimateShearView` call in this block. This way, catching both `GalSimHSMError` and `RuntimeError` with this single block addresses Dan's concerns.

Reply:
That makes sense. The rest of the block is setup code that should not fail and if it does, it's likely a usage error or something we should catch earlier (i.e. a zero-sized image or something like that), not a GalSim/HSM algorithmic failure.